zend/value.cpp

```diff
@@ -335,7 +335,14 @@ Value &Value::operator=(const Value &value)
     // skip self assignment
     if (this == &value) return *this;
 
-    zend_assign_to_variable(_val, value._val, IS_VAR);
+    if (Z_REFCOUNTED_P(_val) || Z_ISREF_P(_val) || Z_ISREF_P(value._val)) {
+        zend_assign_to_variable(_val, value._val, IS_CV);
```

Comment:
I plan to get rid eventually of zend_assign_to_variable() but need to smoke something funny to understand how it works — chances are that `IS_XXX` flag will depend on the type of the variable. Though right now [all my tests](https://github.com/sjinks/PHP-CPP/tree/tests/test) pass, I don't feel confident about it.